utils/config_template.txt

ACME SPD
=========
PDF
---
EVAL_NAME = "ACME_SPD_01" 
EVAL_DIRECTORY = "F:/rag_sdk/datasets/acme_spd/files/pdf" 
EVAL_FILE = "F:/rag_sdk/datasets/acme_spd/files/md/ACME_SPD.pdf"
EVAL_QUESTIONS = "F:/rag_sdk/datasets/acme_spd/questions/ACME_SPD_01_QA.json"
EVAL_RESULTS_DIR = "F:/rag_sdk/evaluations/data/acme_spd"
EVAL_QUICK_TEST = "Are bifocals covered?"

Markdown
--------
EVAL_NAME = "ACME_SPD_01" 
EVAL_DIRECTORY = "F:/rag_sdk/datasets/acme_spd/files/md" 
EVAL_FILE = "F:/rag_sdk/datasets/acme_spd/files/pdf/ACME_SPD.md"
EVAL_QUESTIONS = "F:/rag_sdk/datasets/acme_spd/questions/ACME_SPD_01_QA.xlsx"
EVAL_RESULTS_DIR = "F:/rag_sdk/evaluations/data/acme_spd"
EVAL_QUICK_TEST = "Are bifocals covered?"
EVAL_DB = "F:/rag_sdk/datasets/acme_spd/db/ACME_SPD_01_007_OAI_3_LARGE"
EVAL_PROMPTS_DIR = "F:/rag_sdk/datasets/acme_spd/prompts" 

analysis_file = "F:/rag_sdk/evaluations/diagnostics/acme_spd/ACME_SPD_chunk_analysis.xlsx"

S007_Automated
-------------------
"other_configurations": {
    "EVAL_NAME": "ACME_SPD_01",
    "EVAL_DIRECTORY": "F:/rag_sdk/datasets/acme_spd/files/md",
    "EVAL_FILE": "F:/rag_sdk/datasets/acme_spd/files/pdf/ACME_SPD.md",
    "EVAL_QUESTIONS": "F:/rag_sdk/datasets/acme_spd/questions/ACME_SPD_01_QA.xlsx",
    "EVAL_RESULTS_DIR": "F:/rag_sdk/evaluations/data/acme_spd",
    "EVAL_QUICK_TEST": "Are bifocals covered?",
    "EVAL_DB": "F:/rag_sdk/datasets/acme_spd/db/ACME_SPD_01_007_OAI_3_LARGE",
    "RAG_STRATEGY": "S007_00",
    "SIMILARITY_TOP_K": 30,
    "HEADER_LEVELS": 5,
    "RAG_PROMPT_TEMPLATE": "Your prompt template here",
    "EVALUATION_LLM_FAMILY": "OPENAI",
    "EVALUATION_LLM_MODEL": "gpt-4-0125-preview",
    "EMBEDDING_DIMENSIONS": 768
  }

Chunk Analyzer
--------------
input_file = "F:/rag_sdk/datasets/acme_spd/files/md/ACME_SPD.md"
chunked_file = "F:/rag_sdk/evaluations/diagnostics/acme_spd/ACME_SPD_chunked.md"
analysis_file = "F:/rag_sdk/evaluations/diagnostics/acme_spd/ACME_SPD_chunk_analysis.xlsx"

Albertsons Union Benefits
=========================
PDF
----
EVAL_NAME = "ALB_UNION_BEN_01" 
EVAL_DIRECTORY = "F:/rag_sdk/datasets/alb_union_ben/files/pdf" 
EVAL_FILE = "F:/rag_sdk/datasets/alb_union_ben/files/md/ALB_UNION_BEN.pdf"
EVAL_QUESTIONS = "F:/rag_sdk/datasets/alb_union_ben/questions/ALB_UNION_BEN_01_QA.json"
EVAL_RESULTS_DIR = "F:/rag_sdk/evaluations/data/alb_union_ben"
EVAL_QUICK_TEST = "Are bifocals covered?"

Markdown
----------
EVAL_NAME = "ALB_UNION_BEN_01" 
EVAL_DIRECTORY = "F:/rag_sdk/datasets/alb_union_ben/files/md" 
EVAL_FILE = "F:/rag_sdk/datasets/alb_union_ben/files/md/ALB_UNION_BEN.md"
EVAL_QUESTIONS = "F:/rag_sdk/datasets/alb_union_ben/questions/ALB_UNION_BEN_01_QA.xlsx"
EVAL_RESULTS_DIR = "F:/rag_sdk/evaluations/data/alb_union_ben"
EVAL_QUICK_TEST = "Are bifocals covered?"
EVAL_DB = "F:/rag_sdk/datasets/alb_union_ben/db/ALB_UNION_BEN_01_007_OAI_3_LARGE"
EVAL_PROMPTS_DIR = "F:/rag_sdk/datasets/alb_union_ben/prompts" 

Chunk Analyzer
--------------
input_file = "F:/rag_sdk/datasets/alb_union_ben/files/md/ALB_UNION_BEN.md"
chunked_file = "F:/rag_sdk/evaluations/diagnostics/alb_union_ben/ALB_UNION_BEN_chunked.md"
analysis_file = "F:/rag_sdk/evaluations/diagnostics/alb_union_ben/ALB_UNION_BEN_chunk_analysis.xlsx"

Albertsons Non Union Benefits
=========================
PDF
----
EVAL_NAME = "ALB_NON_UNION_BEN_01" 
EVAL_DIRECTORY = "F:/rag_sdk/datasets/alb_non_union_ben/files/pdf" 
EVAL_FILE = "F:/rag_sdk/datasets/alb_non_union_ben/files/pdf/ALB_NON_UNION_BEN.pdf"
EVAL_QUESTIONS = "F:/rag_sdk/datasets/alb_non_union_ben/questions/ALB_NON_UNION_BEN_01_QA.json"
EVAL_RESULTS_DIR = "F:/rag_sdk/evaluations/data/alb_non_union_ben"
EVAL_QUICK_TEST = "Are bifocals covered?"

Markdown
----------
EVAL_NAME = "ALB_NON_UNION_BEN_01" 
EVAL_DIRECTORY = "F:/rag_sdk/datasets/alb_non_union_ben/files/md" 
EVAL_FILE = "F:/rag_sdk/datasets/alb_non_union_ben/files/md/ALB_NON_UNION_BEN.md"
EVAL_QUESTIONS = "F:/rag_sdk/datasets/alb_non_union_ben/questions/ALB_NON_UNION_BEN_01_QA.xlsx"
EVAL_RESULTS_DIR = "F:/rag_sdk/evaluations/data/alb_non_union_ben"
EVAL_QUICK_TEST = "Are bifocals covered?"
EVAL_DB = "F:/rag_sdk/datasets/alb_non_union_ben/db/ALB_NON_UNION_BEN_01_007_OAI_3_LARGE"
EVAL_PROMPTS_DIR = "F:/rag_sdk/datasets/alb_non_union_ben/prompts" 


Chunk Analyzer
--------------
input_directory = "F:/rag_sdk/datasets/alb_non_union_ben/files/md"
#input_file = "F:/rag_sdk/datasets/alb_non_union_ben/files/md/ALB_NON_UNION_BEN.md"
chunked_file = "F:/rag_sdk/evaluations/diagnostics/alb_non_union_ben/ALB_NON_UNION_BEN_chunked.md"
analysis_file = "F:/rag_sdk/evaluations/diagnostics/alb_non_union_ben/ALB_NON_UNION_BEN_chunk_analysis.xlsx"

ITPEU SPD
=========
PDF
---
EVAL_NAME = "ITPEU_SPD_01" 
EVAL_DIRECTORY = "F:/rag_sdk/datasets/itpeu_spd/files/pdf" 
EVAL_FILE = "F:/rag_sdk/datasets/alb_non_union_ben/files/pdf/ITPEU_SPD.pdf"
EVAL_QUESTIONS = "F:/rag_sdk/datasets/itpeu_spd/questions/ITPEU_SPD_01_QA.json"
EVAL_RESULTS_DIR = "F:/rag_sdk/evaluations/data/itpeu_spd"
EVAL_QUICK_TEST = "Are bifocals covered?"

Markdown
----------
EVAL_NAME = "ITPEU_SPD_01" 
EVAL_DIRECTORY = "F:/rag_sdk/datasets/itpeu_spd/files/md" 
EVAL_FILE = "F:/rag_sdk/datasets/itpeu_spd/files/md/ITPEU_SPD.md"
EVAL_QUESTIONS = "F:/rag_sdk/datasets/itpeu_spd/questions/ITPEU_SPD_01_QA.xlsx"
EVAL_RESULTS_DIR = "F:/rag_sdk/evaluations/data/itpeu_spd"
EVAL_QUICK_TEST = "Are bifocals covered?"
EVAL_DB = "F:/rag_sdk/datasets/itpeu_spd/db/ITPEU_SPD_01_01_007_OAI_3_LARGE"
EVAL_PROMPTS_DIR = "F:/rag_sdk/datasets/itpeu_spd/prompts" 

Chunk Analyzer
--------------
input_file = "F:/rag_sdk/datasets/itpeu_spd/files/md/ITPEU_SPD.md"
chunked_file = "F:/rag_sdk/evaluations/diagnostics/itpeu_spd/ITPEU_SPD_chunked.md"
analysis_file = "F:/rag_sdk/evaluations/diagnostics/itpeu_spd/ITPEU_SPD_chunk_analysis.xlsx"

Kaiser NW Plan
==============
PDF
---
EVAL_NAME = "KAI_NW_PLAN_01" 
EVAL_DIRECTORY = "F:/rag_sdk/datasets/kai_nw_plan/files/pdf" 
EVAL_FILE = "F:/rag_sdk/datasets/kai_nw_plan/files/pdf/KAI_NW_PLAN.pdf"
EVAL_QUESTIONS = "F:/rag_sdk/datasets/kai_nw_plan/questions/KAI_NW_PLAN_01_QA.json"
EVAL_RESULTS_DIR = "F:/rag_sdk/evaluations/data/kai_nw_plan"
EVAL_QUICK_TEST = "Are bifocals covered?"

Markdown
----------
EVAL_NAME = "KAI_NW_PLAN_01" 
EVAL_DIRECTORY = "F:/rag_sdk/datasets/kai_nw_plan/files/md" 
EVAL_FILE = "F:/rag_sdk/datasets/kai_nw_plan/files/md/KAI_NW_PLAN.md"
EVAL_QUESTIONS = "F:/rag_sdk/datasets/kai_nw_plan/questions/KAI_NW_PLAN_01_QA.xlsx"
EVAL_RESULTS_DIR = "F:/rag_sdk/evaluations/data/kai_nw_plan"
EVAL_QUICK_TEST = "Are bifocals covered?"

EVAL_DB = "F:/rag_sdk/datasets/kai_nw_plan/db/KAI_NW_PLAN_01_007_OAI_3_LARGE"
EVAL_PROMPTS_DIR = "F:/rag_sdk/datasets/kai_nw_plan/prompts" 

Chunk Analyzer
--------------
input_file = "F:/rag_sdk/datasets/kai_nw_plan/files/md/KAI_NW_PLAN.md"
chunked_file = "F:/rag_sdk/evaluations/diagnostics/kai_nw_plan/KAI_NW_PLAN_chunked.md"
analysis_file = "F:/rag_sdk/evaluations/diagnostics/kai_nw_plan/KAI_NW_PLAN_chunk_analysis.xlsx"

Cmp Leave
=========
PDF
---
EVAL_NAME = "CMP_LEAVE_01" 
EVAL_DIRECTORY = "F:/rag_sdk/datasets/cmp_leave/files/pdf" 
EVAL_FILE = "F:/rag_sdk/datasets/cmp_leave/files/pdf/"
EVAL_QUESTIONS = "F:/rag_sdk/datasets/cmp_leave/questions/CMP_LEAVE_01_QA.json"
EVAL_RESULTS_DIR = "F:/rag_sdk/evaluations/data/cmp_leave"
EVAL_QUICK_TEST = "Can I take time off to care for my cousin?"

Markdown
--------
EVAL_NAME = "CMP_LEAVE_01" 
EVAL_DIRECTORY = "F:/rag_sdk/datasets/cmp_leave/files/md" 
EVAL_FILE = "F:/rag_sdk/datasets/cmp_leave/files/pdf/"
EVAL_QUESTIONS = "F:/rag_sdk/datasets/cmp_leave/questions/CMP_LEAVE_01_QA.xlsx"
EVAL_RESULTS_DIR = "F:/rag_sdk/evaluations/data/cmp_leave"
EVAL_QUICK_TEST = "Can I take time off to care for my cousin?"
EVAL_DB = "F:/rag_sdk/datasets/cmp_leave/db/CMP_LEAVE_01_OAI_3_LARGE"
EVAL_PROMPTS_DIR = "F:/rag_sdk/datasets/cmp_leave/prompts" 

Chunk Analyzer
----------------
input_directory = "F:/rag_sdk/datasets/cmp_leave/files/md"
#input_file = "F:/rag_sdk/datasets/cmp_leave/files/md/KAI_NW_PLAN.md"
chunked_file = "F:/rag_sdk/evaluations/diagnostics/cmp_leave/CMP_LEAVE_chunked.md"
analysis_file = "F:/rag_sdk/evaluations/diagnostics/cmp_leave/CMP_LEAVE_chunk_analysis.xlsx"

Cmp KA
===============
EVAL_NAME = "CMP_KA_01" 
EVAL_DIRECTORY = "F:/rag_sdk/datasets/cmp_ka/files/md" 
EVAL_FILE = "F:/rag_sdk/datasets/cmp_ka/files/md/*.md"
EVAL_QUESTIONS = "F:/rag_sdk/datasets/cmp_ka/questions/CMP_KA_01_QA.xlsx"
EVAL_RESULTS_DIR = "F:/rag_sdk/evaluations/data/cmp_ka"
EVAL_QUICK_TEST = "How do I create my first job requision?"

EVAL_DB = "F:/rag_sdk/datasets/cmp_ka/db/CMP_KA_01_007_OAI_3_LARGE"
#EVAL_DB = "F:/rag_sdk/datasets/cmp_ka/db/CMP_KA_01_007_COH_3"
EVAL_PROMPTS_DIR = "F:/rag_sdk/datasets/cmp_ka/prompts" 


Cmp OT Emp
=========
PDF
---
EVAL_NAME = "CMP_OT_EMP_01" 
EVAL_DIRECTORY = "F:/rag_sdk/datasets/cmp_ot_emp/files/pdf" 
EVAL_FILE = "F:/rag_sdk/datasets/cmp_ot_emp/files/pdf/"
EVAL_QUESTIONS = "F:/rag_sdk/datasets/cmp_ot_emp/questions/CMP_OT_EMP_01_QA.json"
EVAL_RESULTS_DIR = "F:/rag_sdk/evaluations/data/cmp_ot_emp"
EVAL_QUICK_TEST = "I'm part time. Is overtime based on what I normally work?"

Markdown
--------
EVAL_NAME = "CMP_OT_EMP_01" 
EVAL_DIRECTORY = "F:/rag_sdk/datasets/cmp_ot_emp/files/md" 
EVAL_FILE = "F:/rag_sdk/datasets/cmp_ot_emp/files/md/"
EVAL_QUESTIONS = "F:/rag_sdk/datasets/cmp_ot_emp/questions/CMP_OT_EMP_01_QA.xlsx"
EVAL_RESULTS_DIR = "F:/rag_sdk/evaluations/data/cmp_ot_emp"
EVAL_QUICK_TEST = "I'm part time. Is overtime based on what I normally work?"
EVAL_DB = "F:/rag_sdk/datasets/cmp_ot_emp/db/CMP_OT_EMP_01_OAI_3_LARGE"
#EVAL_DB = "F:/rag_sdk/datasets/cmp_ot_emp/db/CMP_OT_EMP_01_COH_3"
EVAL_PROMPTS_DIR = "F:/rag_sdk/datasets/cmp_ot_emp/prompts" 

Chunk Analyzer
----------------
input_directory = "F:/rag_sdk/datasets/cmp_ot_emp/files/md"
#input_file = "F:/rag_sdk/datasets/cmp_ot_emp/files/md/KAI_NW_PLAN.md"
chunked_file = "F:/rag_sdk/evaluations/diagnostics/cmp_ot_emp/CMP_OT_EMP_chunked.md"
analysis_file = "F:/rag_sdk/evaluations/diagnostics/cmp_ot_emp/CMP_OT_EMP_chunk_analysis.xlsx"


Cmp 401K
=========
PDF
---
EVAL_NAME = "CMP_401K_01" 
EVAL_DIRECTORY = "F:/rag_sdk/datasets/cmp_401K/files/pdf" 
EVAL_FILE = "F:/rag_sdk/datasets/cmp_401K/files/pdf/"
EVAL_QUESTIONS = "F:/rag_sdk/datasets/cmp_401K/questions/CMP_401K_01_QA.json"
EVAL_RESULTS_DIR = "F:/rag_sdk/evaluations/data/cmp_401K"
EVAL_QUICK_TEST = "How do I enroll in the 401k plan?"

Markdown
--------
EVAL_NAME = "CMP_401K_01" 
EVAL_DIRECTORY = "F:/rag_sdk/datasets/cmp_401K/files/md" 
EVAL_FILE = "F:/rag_sdk/datasets/cmp_401K/files/md/"
EVAL_QUESTIONS = "F:/rag_sdk/datasets/cmp_401K/questions/CMP_401K_01_QA.xlsx"
EVAL_RESULTS_DIR = "F:/rag_sdk/evaluations/data/cmp_401K"
EVAL_QUICK_TEST = "How do I enroll in the 401k plan?"
EVAL_DB = "F:/rag_sdk/datasets/cmp_401K/db/CMP_401K_01_OAI_3_LARGE"
#EVAL_DB = "F:/rag_sdk/datasets/cmp_401K/db/CMP_401K_01_COH_3"
EVAL_PROMPTS_DIR = "F:/rag_sdk/datasets/cmp_401K/prompts" 

Chunk Analyzer
----------------
input_directory = "F:/rag_sdk/datasets/cmp_401K/files/md"
chunked_file = "F:/rag_sdk/evaluations/diagnostics/cmp_401K/CMP_401K_chunked.md"
analysis_file = "F:/rag_sdk/evaluations/diagnostics/cmp_401K/CMP_401K_chunk_analysis.xlsx"


Cmp UK RSU
=========
PDF
---
EVAL_NAME = "CMP_UK_RSU_01" 
EVAL_DIRECTORY = "F:/rag_sdk/datasets/cmp_uk_rsu/files/pdf" 
EVAL_FILE = "F:/rag_sdk/datasets/cmp_uk_rsu/files/pdf/"
EVAL_QUESTIONS = "F:/rag_sdk/datasets/cmp_uk_rsu/questions/CMP_UK_RSU_01_QA.json"
EVAL_RESULTS_DIR = "F:/rag_sdk/evaluations/data/cmp_uk_rsu"
EVAL_QUICK_TEST = "What type of taxes do I pay on my RSUs?"

Markdown
--------
EVAL_NAME = "CMP_UK_RSU_01" 
EVAL_DIRECTORY = "F:/rag_sdk/datasets/cmp_uk_rsu/files/md" 
EVAL_FILE = "F:/rag_sdk/datasets/cmp_uk_rsu/files/md/"
EVAL_QUESTIONS = "F:/rag_sdk/datasets/cmp_uk_rsu/questions/CMP_UK_RSU_01_QA.xlsx"
EVAL_RESULTS_DIR = "F:/rag_sdk/evaluations/data/cmp_uk_rsu"
EVAL_QUICK_TEST = "What type of taxes do I pay on my RSUs?"
EVAL_DB = "F:/rag_sdk/datasets/cmp_uk_rsu/db/CMP_UK_RSU_01_OAI_3_LARGE"
#EVAL_DB = "F:/rag_sdk/datasets/cmp_uk_rsu/db/CMP_UK_RSU_01_COH_3"
EVAL_PROMPTS_DIR = "F:/rag_sdk/datasets/cmp_uk_rsu/prompts" 

Chunk Analyzer
----------------
input_directory = "F:/rag_sdk/datasets/cmp_uk_rsu/files/md"
chunked_file = "F:/rag_sdk/evaluations/diagnostics/cmp_uk_rsu/CMP_UK_RSU_chunked.md"
analysis_file = "F:/rag_sdk/evaluations/diagnostics/cmp_uk_rsu/CMP_UK_RSU_chunk_analysis.xlsx"

Cmp NON US RSU
=========
PDF
---
EVAL_NAME = "CMP_NONUS_RSU_01" 
EVAL_DIRECTORY = "F:/rag_sdk/datasets/cmp_nonus_rsu/files/pdf" 
EVAL_FILE = "F:/rag_sdk/datasets/cmp_nonus_rsu/files/pdf/"
EVAL_QUESTIONS = "F:/rag_sdk/datasets/cmp_nonus_rsu/questions/CMP_NONUS_RSU_01_QA.json"
EVAL_RESULTS_DIR = "F:/rag_sdk/evaluations/data/cmp_nonus_rsu"
EVAL_QUICK_TEST = "What happens when my RSUs vest?"

Markdown
--------
EVAL_NAME = "CMP_NONUS_RSU_01" 
EVAL_DIRECTORY = "F:/rag_sdk/datasets/cmp_nonus_rsu/files/md" 
EVAL_FILE = "F:/rag_sdk/datasets/cmp_nonus_rsu/files/md/"
EVAL_QUESTIONS = "F:/rag_sdk/datasets/cmp_nonus_rsu/questions/CMP_NONUS_RSU_01_QA.xlsx"
EVAL_RESULTS_DIR = "F:/rag_sdk/evaluations/data/cmp_nonus_rsu"
EVAL_QUICK_TEST = "What happens when my RSUs vest?"
EVAL_DB = "F:/rag_sdk/datasets/cmp_nonus_rsu/db/CMP_NONUS_RSU_01_OAI_3_LARGE"
#EVAL_DB = "F:/rag_sdk/datasets/cmp_nonus_rsu/db/CMP_NONUS_RSU_01_COH_3"
EVAL_PROMPTS_DIR = "F:/rag_sdk/datasets/cmp_nonus_rsu/prompts" 

Chunk Analyzer
----------------
input_directory = "F:/rag_sdk/datasets/cmp_nonus_rsu/files/md"
chunked_file = "F:/rag_sdk/evaluations/diagnostics/cmp_nonus_rsu/CMP_NONUS_RSU_chunked.md"
analysis_file = "F:/rag_sdk/evaluations/diagnostics/cmp_nonus_rsu/CMP_NONUS_RSU_chunk_analysis.xlsx"


Cmp Proc Guide
=========
PDF
---
EVAL_NAME = "CMP_PROC_GUIDE_01" 
EVAL_DIRECTORY = "F:/rag_sdk/datasets/cmp_proc_guide/files/pdf" 
EVAL_FILE = "F:/rag_sdk/datasets/cmp_proc_guide/files/pdf/"
EVAL_QUESTIONS = "F:/rag_sdk/datasets/cmp_proc_guide/questions/CMP_PROC_GUIDE_01_QA.json"
EVAL_RESULTS_DIR = "F:/rag_sdk/evaluations/data/cmp_proc_guide"
EVAL_QUICK_TEST = "One of the sponsors I am using has some questions around PII security, what should I do?"

Markdown
--------
EVAL_NAME = "CMP_PROC_GUIDE_01" 
EVAL_DIRECTORY = "F:/rag_sdk/datasets/cmp_proc_guide/files/md" 
EVAL_FILE = "F:/rag_sdk/datasets/cmp_proc_guide/files/md/"
EVAL_QUESTIONS = "F:/rag_sdk/datasets/cmp_proc_guide/questions/CMP_PROC_GUIDE_01_QA.xlsx"
EVAL_RESULTS_DIR = "F:/rag_sdk/evaluations/data/cmp_proc_guide"
EVAL_QUICK_TEST = "One of the sponsors I am using has some questions around PII security, what should I do?"
EVAL_DB = "F:/rag_sdk/datasets/cmp_proc_guide/db/CMP_PROC_GUIDE_01_OAI_3_LARGE"
#EVAL_DB = "F:/rag_sdk/datasets/cmp_proc_guide/db/CMP_PROC_GUIDE_01_COH_3"
EVAL_PROMPTS_DIR = "F:/rag_sdk/datasets/cmp_proc_guide/prompts" 

Chunk Analyzer
----------------
input_directory = "F:/rag_sdk/datasets/cmp_proc_guide/files/md"
chunked_file = "F:/rag_sdk/evaluations/diagnostics/cmp_proc_guide/CMP_PROC_GUIDE_chunked.md"
analysis_file = "F:/rag_sdk/evaluations/diagnostics/cmp_proc_guide/CMP_PROC_GUIDE_chunk_analysis.xlsx"

SOW CBA
=======

PDF
---
EVAL_NAME = "SOW_CBA_01" 
EVAL_DIRECTORY = "F:/rag_sdk/datasets/sow_cba/files/pdf" 
EVAL_FILE = "F:/rag_sdk/datasets/sow_cba/files/pdf/"
EVAL_QUESTIONS = "F:/rag_sdk/datasets/sow_cba/questions/SOW_CBA_01_QA.json"
EVAL_RESULTS_DIR = "F:/rag_sdk/evaluations/data/sow_cba"
EVAL_QUICK_TEST = "Who is my employer?"

Markdown
--------
EVAL_NAME = "SOW_CBA_01" 
EVAL_DIRECTORY = "F:/rag_sdk/datasets/sow_cba/files/md" 
EVAL_FILE = "F:/rag_sdk/datasets/sow_cba/files/md/"
EVAL_QUESTIONS = "F:/rag_sdk/datasets/sow_cba/questions/SOW_CBA_01_QA.xlsx"
EVAL_RESULTS_DIR = "F:/rag_sdk/evaluations/data/sow_cba"
EVAL_QUICK_TEST = "Who is my employer?"
EVAL_DB = "F:/rag_sdk/datasets/sow_cba/db/SOW_CBA_01_OAI_3_LARGE"
#EVAL_DB = "F:/rag_sdk/datasets/sow_cba/db/SOW_CBA_01_COH_3"
EVAL_PROMPTS_DIR = "F:/rag_sdk/datasets/sow_cba/prompts" 

Chunk Analyzer
----------------
input_directory = "F:/rag_sdk/datasets/sow_cba/files/md"
chunked_file = "F:/rag_sdk/evaluations/diagnostics/sow_cba/SOW_CBA_chunked.md"
analysis_file = "F:/rag_sdk/evaluations/diagnostics/sow_cba/SOW_CBA_chunk_analysis.xlsx"


ISO_13485_2016
================

Markdown
--------
EVAL_NAME = "ISO_13485_2016_01" 
EVAL_DIRECTORY = "F:/rag_sdk/datasets/ISO_13485_2016/files/md" 
EVAL_FILE = "F:/rag_sdk/datasets/ISO_13485_2016/files/md/"
EVAL_QUESTIONS = "F:/rag_sdk/datasets/ISO_13485_2016/questions/ISO_13485_2016_01_QA.xlsx"
EVAL_RESULTS_DIR = "F:/rag_sdk/evaluations/data/ISO_13485_2016"
EVAL_QUICK_TEST = "What do I need to document for each medical device?"
EVAL_DB = "F:/rag_sdk/datasets/ISO_13485_2016/db/ISO_13485_2016_S007_OAI_3_LARGE"
#EVAL_DB = "F:/rag_sdk/datasets/ISO_13485_2016/db/ISO_13485_2016_S007_COH_3"
EVAL_PROMPTS_DIR = "F:/rag_sdk/datasets/sow_cba/prompts" 


DELL_I14_COMPLIANCE
================

Markdown
--------
EVAL_NAME = "DELL_I14_COMPLIANCE_01" 
EVAL_DIRECTORY = "F:/rag_sdk/datasets/dell_i14_compliance/files/md" 
EVAL_FILE = "F:/rag_sdk/datasets/dell_i14_compliance/files/md/*.md"
EVAL_QUESTIONS = "F:/rag_sdk/datasets/dell_i14_compliance/questions/DELL_I14_COMPLIANCE_01_QA.xlsx"
EVAL_RESULTS_DIR = "F:/rag_sdk/evaluations/data/dell_i14_compliance"
EVAL_QUICK_TEST = "What safety certifications does the Dell Inspiron 14 comply with?"
EVAL_DB = "F:/rag_sdk/datasets/dell_i14_compliance/db/DELL_I14_COMPLIANCE_01_007_OAI_3_LARGE"
#EVAL_DB = "F:/rag_sdk/datasets/dell_i14_compliance/db/DELL_I14_COMPLIANCE_01_007_COH_3"
EVAL_PROMPTS_DIR = "F:/rag_sdk/datasets/sow_cba/prompts" 


CMP_OCAM
================

Markdown
--------
EVAL_NAME = "CMP_OCAM_01" 
EVAL_DIRECTORY = "F:/rag_sdk/datasets/cmp_ocam/files/md" 
EVAL_FILE = "F:/rag_sdk/datasets/cmp_ocam/files/md/*.md"
EVAL_QUESTIONS = "F:/rag_sdk/datasets/cmp_ocam/questions/CMP_OCAM_01_QA.xlsx"
EVAL_RESULTS_DIR = "F:/rag_sdk/evaluations/data/cmp_ocam"
EVAL_QUICK_TEST = "What is the Direct Approval Chain?"
EVAL_DB = "F:/rag_sdk/datasets/cmp_ocam/db/CMP_OCAM_01_007_OAI_3_LARGE"
#EVAL_DB = "F:/rag_sdk/datasets/cmp_ocam/db/DCMP_OCAM_01_007_COH_3"
EVAL_PROMPTS_DIR = "F:/rag_sdk/datasets/sow_cba/prompts" 

CMP_HOSTING
================

Markdown
--------
EVAL_NAME = "CMP_HOSTING_01" 
EVAL_DIRECTORY = "F:/rag_sdk/datasets/cmp_hosting/files/md" 
EVAL_FILE = "F:/rag_sdk/datasets/cmp_hosting/files/md/*.md"
EVAL_QUESTIONS = "F:/rag_sdk/datasets/cmp_hosting/questions/CMP_HOSTING_01_QA.xlsx"
EVAL_RESULTS_DIR = "F:/rag_sdk/evaluations/data/cmp_hosting"
EVAL_QUICK_TEST = "My company is in Italy, which data center region is my data hosted in?"
EVAL_DB = "F:/rag_sdk/datasets/cmp_hosting/db/CMP_HOSTING_01_007_OAI_3_LARGE"
#EVAL_DB = "F:/rag_sdk/datasets/cmp_hosting/db/CMP_HOSTING_01_007_COH_3"
EVAL_PROMPTS_DIR = "F:/rag_sdk/datasets/cmp_hosting/prompts" 





config.py
import os

#OPENAI_API_KEY = ""
#TOGETHER_API_KEY = ""
#LLAMA_CLOUD_API_KEY = ""
#COHERE_API_KEY = ""
#ANTHROPIC_API_KEY = ""
#GOOGLE_API_KEY = ""
#LANGCHAIN_API_KEY = ""


EVAL_NAME = "CMP_HOSTING_01" 
EVAL_DIRECTORY = "F:/rag_sdk/datasets/cmp_hosting/files/md" 
EVAL_FILE = "F:/rag_sdk/datasets/cmp_hosting/files/md/*.md"
EVAL_QUESTIONS = "F:/rag_sdk/datasets/cmp_hosting/questions/CMP_HOSTING_01_QA.xlsx"
EVAL_RESULTS_DIR = "F:/rag_sdk/evaluations/data/cmp_hosting"
EVAL_QUICK_TEST = "My company is in Italy, which data center region is my data hosted in?"
EVAL_DB = "F:/rag_sdk/datasets/cmp_hosting/db/CMP_HOSTING_01_007_OAI_3_LARGE"
#EVAL_DB = "F:/rag_sdk/datasets/cmp_hosting/db/CMP_HOSTING_01_007_COH_3"
EVAL_PROMPTS_DIR = "F:/rag_sdk/datasets/cmp_hosting/prompts" 
 


# OPENAI, COHERE, ANTHROPIC, GOOGLE, META, QWEN, MISTRALAI
GENERATION_LLM_FAMILY = "OPENAI" 

# gpt-4o, command-r, command-r-plus, claude-3-5-sonnet-20240620, models/gemini-1.5-pro, 
# meta-llama/Llama-3-70b-chat-hf, meta-llama/Meta-Llama-3.1-405B-Instruct-Turbo, meta-llama/Meta-Llama-3.1-70B-Instruct-Turbo, 
# Qwen/Qwen2-72B-Instruct, mistralai/Mixtral-8x22B-Instruct-v0.1
#accounts/fireworks/models/llama-v3p1-405b-instruct, accounts/fireworks/models/llama-v3p1-70b-instruct

GENERATION_LLM_MODEL = "gpt-4o"

# OPENAI, COHERE, GOOGLE
EMBEDDING_LLM_FAMILY = "OPENAI" 

# text-embedding-3-large, embed-english-v3.0, models/text-embedding-004
EMBEDDING_LLM_MODEL = "text-embedding-3-large" 
EMBEDDING_DIMESIONS = 1024

EVALUATION_LLM_FAMILY = "OPENAI" # OPENAI, COHERE
EVALUATION_LLM_MODEL = "gpt-4-0125-preview" # gpt-4o, gpt-4-0125-preview

RAG_STRATEGY = "S007_00"

# Common Setting
CHUNK_SIZE = 512
SIMILARITY_TOP_K = 30
SIMILARITY_CUTOFF = 0.2 

#Reranker settings
RERANKER = "COHERE" #Will add others later
RERANKER_MODEL = "rerank-english-v3.0"
RERANK_TOP_N = 10

# S002 -> Sentence Window Retriever Settings
WINDOW_SIZE = 5

# S003 -> Recursive Retriever Settings
PARENT_CHUNK_SIZE = 1024
SUB_CHUNK_SIZES = "128_256_512"

# S004 -> Fusion Retriever Settings
RETRIEVER_WEIGHTS = "0.5_0.5"
FUSION_RERANKER = "reciprocal_rerank" # reciprocal_rerank, relative_score, dist_based_score, simple

#S007 -> Semantic
HEADER_LEVELS = 5

#S008 -> Black Box 
BB_OUTPUT_FILE = "F:/rag_sdk/evaluations/black_box/ALB_NON_UNION_BEN_01_FAI_07_30.xlsx"

#S009 -> Retriever 
RTR_OUTPUT_FILE = ""

# S010 -> LC Ensemble Retriever
LC_ENSEMBLE_WEIGHTS = "1.0_0.0"


# Standard Langchain prompt
rag_prompt_lc_01 = """You are an assistant for question-answering tasks. 
Use the retrieved context, consisting of these documents, to answer the question. 
If you don't know the answer, just say that you don't know. 
Provide a detailed response, but do not invent stuff
\nContext: {context}
\nQuestion: {question}
"""

# Standard Cohere preamble
rag_prompt_coh_01 = """## Task & Context
You are an expert Human Resources assistant that helps employees answer questions about company policies. \
Use the provided documents to answer questions about an employee's specific situation.

## Style Guide
- Think step by step, provide evidence and/or reasoning first, then the answer."""

RAG_PROMPT_TEMPLATE = rag_prompt_lc_01

def set_environment():
    variable_dict = globals().items()
    for key, value in variable_dict:
        #if "API" in key or "ID" in key:
        os.environ[key] = str(value)



